addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4577,157 +4812,136 @@ registry.ColoredLevelBackground = registry.BackgroundToggler.extend({
  * @todo replace this mechanism with real backend m2o field ?
  */
 registry.many2one = SnippetOptionWidget.extend({
-    xmlDependencies: ['/web_editor/static/src/xml/snippets.xml'],
+    display_fields: {
+        // Currently unused, only display_name is displayed, name is used when
+        // rendering into the page. Not sure about how to pass-in a way to format
+        // the displayed record in the select. Maybe they can pass a template id?
+        // Not sure how useful it is to keep since we now have working preview.
+        'res.partner': ['name', 'display_name', 'city', 'country_id'],
+    },
+
     /**
      * @override
      */
-    start: function () {
-        var self = this;
+    async willStart() {
+        const _super = this._super.bind(this, ...arguments);
+        const {oeMany2oneModel, oeMany2oneId} = this.$target[0].dataset;
+        this.fields = this.display_fields[oeMany2oneModel] || ['name', 'display_name'];
+        this.initialRecord = (await this._rpc({
+            model: oeMany2oneModel,
+            method: 'read',
+            args: [[parseInt(oeMany2oneId)], this.fields],
+        }))[0];
+        return _super();
+    },
```

Comment:
```suggestion
    willStart() {
        const {oeMany2oneModel, oeMany2oneId} = this.$target[0].dataset;
        this.fields = this.display_fields[oeMany2oneModel] || ['name', 'display_name'];
        this.initialRecord = null;
        const initialRecordFetcher = this._rpc({
            model: oeMany2oneModel,
            method: 'read',
            args: [[parseInt(oeMany2oneId)], this.fields],
        }).then((records) => {
            this.initialRecord = records[0];
        });
        return Promise.all([this._super(...arguments), initialRecordFetcher]);
    },
```
should not block `_super()` on the resolution of `read`, I don't see why there'd be a rependency between the two.